src/client/play.js

```diff
@@ -5,12 +5,14 @@ const concat = require('../transforms/binaryStream').concat
 module.exports = function (client, options) {
   client.once('success', onLogin)
 
+  const EMPTY_BUFFER = Buffer.from([])
+
   function onLogin (packet) {
     client.state = states.PLAY
     client.uuid = packet.uuid
     client.username = packet.username
     client.signMessage = (message, timestamp, salt = 0) => {
-      if (!client.profileKeys) throw Error("Can't sign message without profile keys, please set valid auth mode")
+      if (!client.profileKeys) return EMPTY_BUFFER
```

Comment:
this should be an exception as no signature is returned

Reply:
Fixed in latest commit